src/librustc/mir/interpret/allocation.rs

```diff
@@ -536,6 +578,91 @@ impl<'tcx, Tag, Extra> Allocation<Tag, Extra> {
     }
 }
 
+/// Run-length encoding of the undef mask.
+/// Used to copy parts of a mask multiple times to another allocation.
+pub struct AllocationDefinedness {
+    ranges: smallvec::SmallVec::<[u64; 1]>,
+    first: bool,
```

Comment:
`inital` might be a better field name? Also please add a doc comment, and maybe reorder the field.

My first reaction when I just saw this "why does it matter if this is the first definedness every created...?"